src/components/defectReport/DefectReportLocationForm.tsx

```diff
@@ -0,0 +1,115 @@
+import * as Location from 'expo-location';
+import React, { useCallback, useEffect, useState } from 'react';
+import Collapsible from 'react-native-collapsible';
+
+import { texts } from '../../config';
+import {
+  useLastKnownPosition,
+  useLocationSettings,
+  usePosition,
+  useSystemPermission
+} from '../../hooks';
+import { MapMarker } from '../../types';
+import { Button } from '../Button';
+import { LoadingSpinner } from '../LoadingSpinner';
+import { Map } from '../map/Map';
+import { baseLocationMarker, getLocationMarker } from '../settings';
+import { RegularText } from '../Text';
+import { Touchable } from '../Touchable';
+import { Wrapper } from '../Wrapper';
+
+export const DefectReportLocationForm = ({
+  setIsLocationSelect,
+  selectedPosition,
+  setSelectedPosition
+}: {
+  setIsLocationSelect: (isLocationSelect: boolean) => void;
+  selectedPosition: Location.LocationObjectCoords | undefined;
+  setSelectedPosition: (position: Location.LocationObjectCoords | undefined) => void;
+}) => {
+  const [shouldGetPosition, setShouldGetPosition] = useState(false);
+  const [showMap, setShowMap] = useState(false);
+
+  const { locationSettings } = useLocationSettings();
+  const systemPermission = useSystemPermission();
+  const { loading: loadingPosition, position } = usePosition(!shouldGetPosition);
+  const { loading: loadingLastKnownPosition, position: lastKnownPosition } =
+    useLastKnownPosition(shouldGetPosition);
+
+  const onPressPosition = useCallback(async () => {
+    setShouldGetPosition(true);
+  }, []);
+
+  useEffect(() => {
+    if (
+      shouldGetPosition &&
+      !loadingPosition &&
+      !loadingLastKnownPosition &&
+      (position || lastKnownPosition)
+    ) {
+      setIsLocationSelect(false);
+      setSelectedPosition((position || lastKnownPosition)?.coords);
+    }
+  }, [lastKnownPosition, loadingPosition, loadingLastKnownPosition, position, shouldGetPosition]);
```

Comment:
If the location permissions in the system are turned off, the location value is `undefined`. 

Is it supposed to be like this? Or should we ask the user to grant location permission?

Reply:
I think it's better now.